src/main/java/org/springframework/data/tarantool/core/BaseTarantoolTemplate.java

```diff
@@ -360,10 +366,15 @@ protected <T> ValueConverter<Value, List<T>> getListValueConverter(Class<T> enti
         if (!spaceMetadata.isPresent() && !entityClass.equals(void.class)) {
             throw new TarantoolMetadataMissingException(name);
         }
-        return tarantoolClient
-                .getResultMapperFactoryFactory()
-                .defaultTupleSingleResultMapperFactory()
-                .withDefaultTupleValueConverter(mapper, spaceMetadata.orElse(null));
+
+        return mapperFactoryFactory.createMapper(mapper)
+                       .withSingleValueConverter(
+                               mapperFactoryFactory.createMapper(mapper, spaceMetadata.orElse(null))
+                                       .withArrayValueToTarantoolTupleResultConverter()
+                                       .withRowsMetadataToTarantoolTupleResultConverter()
+                                       .buildCallResultMapper(
+                                               DefaultMessagePackMapperFactory.getInstance().emptyMapper()))
+                       .buildCallResultMapper(DefaultMessagePackMapperFactory.getInstance().emptyMapper());
```

Comment:
That empty mapper here looks a bit confusing. Seems that it should be either a default mapper, or no mapper parameter at all (the empty mapper as a default value in the method).

Reply:
It is confusing logic. I suggest fixing it after https://github.com/tarantool/cartridge-springdata/pull/124#discussion_r1182711256

Reply:
Let's add a ticket to the driver at least?